### ISSUE TITLE
Anyone have this working well on M1?

### ISSUE THREAD

**Issue by cantonalex** [chychkan/DeepFaceLab_MacOS#69]
Having issues installing numpy and tensorflow packages on my M1. Interested to know if someone is successfully training.

**Comment by xfl0203** [comment]
you can install tensorflow with this url
https://developer.apple.com/metal/tensorflow-plugin/

**Comment by xfl0203** [comment]
> Having issues installing numpy and tensorflow packages on my M1. Interested to know if someone is successfully training.

i use conda to install numpy
conda install -n "your env`name" numpy

**Comment by chychkan** [comment]
With recent fixes, now there's partial support for M1 laptops. XSeg is still not working (there's separate open issue https://github.com/chychkan/DeepFaceLab_MacOS/issues/77), but the rest of the scripts should be usable now.

**Comment by cantonalex** [comment]
I'm still getting that numpy error any suggestions?
`note: This error originates from a subprocess, and is likely not a problem with pip.
        ERROR: Failed building wheel for numpy
      Failed to build numpy
      ERROR: Could not build wheels for numpy, which is required to install pyproject.toml-based projects
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install backend dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

`

**Comment by chychkan** [comment]
@cantonalex I would suggest starting from scratch, just to avoid any weird leftovers from previous attempts, so clone this repo afresh. If still the same error, then you can also try installing numpy manually by activating the virtual env (command: `source .dfl/env/bin/activate`) and installing numpy with `pip install numpy==1.21.4`. If that is successful, then you can try running the setup script again.

**Comment by cantonalex** [comment]
@chychkan got it working but can't seem to with metal. https://github.com/chychkan/DeepFaceLab_MacOS/issues/81#issuecomment-1059767585

**Comment by chychkan** [comment]
Yeah, there's a problem with metal. Looks like the code of DeepFaceLab itself needs to be adjusted to support GPU on M1's.

**Comment by marshalleq** [comment]
So that means no GPU training?

**Comment by prashantspandey** [comment]
Any updates, did anyone get it to run on GPU?

**Comment by marshalleq** [comment]
Can't get it to run at all on Ventura.
